MaristCampusMap/MaristCampusMap/MainScreenViewController.swift

```diff
@@ -170,35 +178,32 @@ extension MainScreenViewController : UIPickerViewDataSource, UIPickerViewDelegat
     func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
         print(self.pickerOptions[row].name)
         self.selectedLocation = self.pickerOptions[row]
-        //self.SelectedLocationLabel.text = self.selectedLocation!.name
     }
 }
 
+
 extension MainScreenViewController : CLLocationManagerDelegate {
-    //print current location
+    //get user's current location
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        //let locValue:CLLocationCoordinate2D = manager.location!.coordinate
 
         if let location = locations.last {
             
             self.lastLocation = location
-            print(location)
-            //compare selected location to this location and get angle bearing for arrow
-            //then update arrow orientation, compare to self.selected
-            /*let _selectedLocation = CLLocation(latitude: selectedLocation.latitude, longitude: selectedLocation.longitude)
-            //print(location)
-            let bearingInRad = getBearingBetweenTwoPoints1(point1: location, point2: _selectedLocation)
             
-            print(rad2deg(bearingInRad))
-            //MapArrow.transform = CGAffineTransform(rotationAngle: CGFloat(bearingInRad))
-            let latestBearing = CGFloat(bearingInRad)
-            UIView.animate(withDuration: 0.5) {
-            self.MapArrow.transform = CGAffineTransform(rotationAngle: latestBearing - latestHeading)
-            }*/
-
+            if let selectedLocation =
+                self.selectedPickerOption?.location {
+            let _selectedLocation = CLLocation(latitude: selectedLocation.latitude, longitude: selectedLocation.longitude)
```

Comment:
Perhaps you could improve readability by eliminating similar/redundant naming and shortening the code, such as:

```swift
if let selectedOption =
                self.selectedPickerOption? {
            let selectedLocation = CLLocation( latitude: selectedOption.location.latitude, longitude: selectedOption.location.longitude )
            // or the following, if it will work without passing altitude, accuracy and other arguments
            let selectedLocation = CLLocation( coordinate: selectedOption.location )
```

Reply:
Replaced code with the first option, the second one required the other parameters like altitude